pkg/stub/handler.go

```diff
@@ -23,16 +24,42 @@ type Handler struct {
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	if event.Deleted {
-		return nil
-	}
 	switch o := event.Object.(type) {
 	case *dnsv1alpha1.ClusterDNS:
-		return h.syncDNSUpdate(o)
+		if event.Deleted {
+			return h.deleteDNS(o)
```

Comment:
Something else to note about the event-driven delete pattern is that if any step fails, I'm not sure we'll actually get a new delete event dequeued. Just another random aside regarding level driven deletes.